systemd/systemd.go

```diff
@@ -1368,18 +1368,15 @@ const mountUnitTemplate = `[Unit]
 Description=Mount unit for {{.SnapName}}
 {{- with .Revision}}, revision {{.}}{{end}}
 {{- with .Origin}} via {{.}}{{end}}
-Before=snapd.service
-After=zfs-mount.service
+After=snapd.mounts-pre.target
+Before=snapd.mounts.target
 
 [Mount]
 What={{.What}}
 Where={{.Where}}
 Type={{.Fstype}}
 Options={{join .Options ","}}
 LazyUnmount=yes
-
-[Install]
-WantedBy=default.target multi-user.target
```

Comment:
Aren't we missing a dependency between these mount units and the `snapd.mounts.target`? It seems you only added temporal constraint, but no dependencies.

Reply:
Oh right, sorry, it should be `WantedBy=snapd.mounts.target`

Reply:
Fixed